[UI-side compositing] Fix 2-up shrink mode to work with UI-side compositing

nmoucht opened the pull request:
#### f4154cb024ec669f8cd2355ee59896164c2e7a47
<pre>
[UI-side compositing] Fix 2-up shrink mode to work with UI-side compositing
<a href="https://bugs.webkit.org/show_bug.cgi?id=250172">https://bugs.webkit.org/show_bug.cgi?id=250172</a>
&lt;rdar://103373463&gt;

Reviewed by Simon Fraser.

Move TiledCoreAnimationDrawingArea::scaleViewToFitDocumentIfNeeded to DrawingArea
so that it can be called in RemoteLayerTreeDrawingArea as well. Call is added to
RemoteLayerTreeDrawingArea::updateRendering.

* Source/WebKit/WebProcess/WebPage/DrawingArea.cpp:
(WebKit::DrawingArea::setShouldScaleViewToFitDocument):
(WebKit::DrawingArea::scaleViewToFitDocumentIfNeeded):
* Source/WebKit/WebProcess/WebPage/DrawingArea.h:
(WebKit::DrawingArea::setShouldScaleViewToFitDocument): Deleted.
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm:
(WebKit::RemoteLayerTreeDrawingArea::updateRendering):
* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.h:
* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.mm:
(WebKit::TiledCoreAnimationDrawingArea::setShouldScaleViewToFitDocument): Deleted.
(WebKit::TiledCoreAnimationDrawingArea::scaleViewToFitDocumentIfNeeded): Deleted.

Canonical link: <a href="https://commits.webkit.org/258565@main">https://commits.webkit.org/258565@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b7ccbcdc3f4d3e4cabbfd513ae795df4b763c798

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102356 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11492 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35421 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111663 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12493 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2412 "Built successfully") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/94677 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/109368 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108137 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/9563 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/92830 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/94677 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/91431 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/24314 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/94677 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/5001 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25737 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5142 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/2175 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11170 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45230 "Passed tests") | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/74/builds/5890 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6892 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3112 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebKit/WebProcess/WebPage/DrawingArea.h:
```diff
@@ -178,6 +178,10 @@ class DrawingArea : public IPC::MessageReceiver, public WebCore::DisplayRefreshM
     const DrawingAreaType m_type;
     DrawingAreaIdentifier m_identifier;
     WebPage& m_webPage;
+    bool m_shouldScaleViewToFitDocument;
```
Initialize this to `{ false }` and move it next to `m_isScalingViewToFitDocument` to optimize padding.

nmoucht replied:
Fixed.

smfr commented on Source/WebKit/WebProcess/WebPage/DrawingArea.cpp:
```diff
@@ -175,4 +176,106 @@ void DrawingArea::prepopulateRectForZoom(double scale, WebCore::FloatPoint origi
+    if (!m_shouldScaleViewToFitDocument)
+        return;
+
+    LOG(Resize, "TiledCoreAnimationDrawingArea %p scaleViewToFitDocumentIfNeeded", this);
```
Log is wrong now.

nmoucht replied:
Fixed.

nmoucht committed:
[UI-side compositing] Fix 2-up shrink mode to work with UI-side compositing
https://bugs.webkit.org/show_bug.cgi?id=250172
<rdar://103373463>

Reviewed by Simon Fraser.

Move TiledCoreAnimationDrawingArea::scaleViewToFitDocumentIfNeeded to DrawingArea
so that it can be called in RemoteLayerTreeDrawingArea as well. Call is added to
RemoteLayerTreeDrawingArea::updateRendering.

* Source/WebKit/WebProcess/WebPage/DrawingArea.cpp:
(WebKit::DrawingArea::setShouldScaleViewToFitDocument):
(WebKit::DrawingArea::scaleViewToFitDocumentIfNeeded):
* Source/WebKit/WebProcess/WebPage/DrawingArea.h:
(WebKit::DrawingArea::setShouldScaleViewToFitDocument): Deleted.
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm:
(WebKit::RemoteLayerTreeDrawingArea::updateRendering):
* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.h:
* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.mm:
(WebKit::TiledCoreAnimationDrawingArea::setShouldScaleViewToFitDocument): Deleted.
(WebKit::TiledCoreAnimationDrawingArea::scaleViewToFitDocumentIfNeeded): Deleted.

Canonical link: https://commits.webkit.org/258565@main
